File: 26. IsSubsequence.cpp
#include <iostream>
#include <string>
using namespace std;

bool isSubsequence(string s, string t) {
    int i = 0;
    for (int j = 0; j < (int)t.size() && i < (int)s.size(); j++) {
        if (s[i] == t[j]) i++;
    }
    return i == (int)s.size();
}

int main() {
    cout << boolalpha;
    cout << isSubsequence("abc", "ahbgdc") << "\n";
    cout << isSubsequence("axc", "ahbgdc") << "\n";
    cout << isSubsequence("", "abc") << "\n";
}

File: 26. IsSubsequence.java
public class IsSubsequence {

    public static boolean isSubsequence(String s, String t) {
        int i = 0;
        for (int j = 0; j < t.length() && i < s.length(); j++) {
            if (s.charAt(i) == t.charAt(j)) i++;
        }
        return i == s.length();
    }

    public static void main(String[] args) {
        System.out.println(isSubsequence("abc", "ahbgdc")); // true
        System.out.println(isSubsequence("axc", "ahbgdc")); // false
        System.out.println(isSubsequence("", "abc"));       // true
    }
}
